src/main.rs

```diff
@@ -158,3 +178,54 @@ fn main() -> Result<()> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests{
+    use super::*;
+
+    #[test]
+    fn test_pet() -> Result<()>{
+        // Checks that the GPIO is initiallized low,
+        // Checks if it can be set high,
+        // Checks if it can be set back low.
+
+        let mut petter = Petter::new()?;
+        let mut config = petter.hand.config();
+
+        petter.hand.reconfigure(config.as_input())?;
+        let mut line_value = petter.hand.value(petter.lane)?;
```

Comment:
I think I misunderstood the issues you were having with the GPIO library when you mentioned it before, but I understand now. Instead of reading the line value from the GPIO line itself, check that it was set appropriately through `gpio-sim`, from the file you mention down below. This won't conflict with the library taking ownership of the line. This way the line that `Petter` owns doesn't need to be reconfigured, and you shouldn't have to manipulate the `Petter` internals really at all. It's as-if we were looking at the physical outputs with an oscilloscope or something but instead it's all simulated.

This does mean that this test can't be run on the live C3 hardware and that's OK. Verifying that the line is getting set appropriately can be done in another way, probably by looking at the physical lines or something. We can come up with a battery of tests to run on the physical hardware latter, but for now consider it out of scope.

Where I want to run this test is on my personal dev machine (and eventually github runners) with whatever arbitrary distro I'm running. It has some real GPIO lines, but not a `gpiochip2`, and a new enough kernel to have `gpio-sim` available but not currently loaded. What I want is for the test to create a sim GPIO line of the appropriate configuration (or fail and tell me that it can't. Alternatively it might be worthwhile to pass an arbitrary chip/pin to `Petter`?), run some checks that ensure the line is being used correctly, and then tear down the sim line. There's a lot of boilerplate involved in that but that `gpiosim` library I linked a while back should help cut it down.

Because the line shouldn't need to be reconfigured I believe this can be done with the original `gpiod` library, so I'd also like to undo the `gpiocdev` conversion undone. If there is another showstopper issue with `gpiod` the new library can always be returned to.

Reply:
`setup_sims.sh` is a really useful script and I want to use it in a full integration test environment. On the other hand I can't use it locally and it introduces a dependency on another OreSat repo. I'd like to keep this one self contained as much as is reasonable. This project only needs a single line.

>  I don't currently have the hardware to be able to test that

That should be fine, I don't anticipate having live ones in the mix to be difficult to work around. Attempting to create the resource and if it fails letting the user sort it out is probably enough.